vms/rpcchainvm/ghttp/http_server.go

```diff
@@ -142,7 +141,21 @@ func (s *Server) Handle(ctx context.Context, req *httppb.HTTPRequest) (*emptypb.
 
 	s.handler.ServeHTTP(writer, request)
 
-	return &emptypb.Empty{}, clientConn.Close()
+	if err := clientConn.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close client conn: %w", err)
+	}
+
+	header := make([]*httppb.Element, 0, len(writerHeaders))
+	for k, v := range writerHeaders {
+		header = append(header, &httppb.Element{
+			Key:    k,
+			Values: v,
+		})
+	}
```

Comment:
Looks like we already have a helper for this
```suggestion
	header := grpcutils.GetHTTPHeader(writerHeaders)
```